### PR TITLE
Keep invoice/proforma in sync on transaction updates

### PR DESCRIPTION
**Pull request by calind**

### CODE REVIEW

File: silver/models/transactions.py [hunk 1]
```diff
@@ -235,6 +240,11 @@ def pre_transaction_save(sender, instance=None, **kwargs):
     if not getattr(instance, 'cleaned', False):
         instance.full_clean()
 
+    transaction = instance
```
Comment (vtemian) [inline silver/models/transactions.py]: You can move this one as the first line in this function.

File: silver/models/transactions.py [hunk 2]
```diff
@@ -235,6 +240,11 @@ def pre_transaction_save(sender, instance=None, **kwargs):
...
+    proforma = transaction.proforma
+    if proforma and proforma.invoice and transaction.invoice is None:
```
Comment (vtemian) [inline silver/models/transactions.py]: `if proforma and proforma.invoice and not transaction.invoice:`

File: silver/tests/factories.py [hunk 3]
```diff
@@ -247,6 +247,23 @@ class Meta:
     customer = factory.SubFactory(CustomerFactory)
 
 
+class ProformaTransactionFactory(factory.django.DjangoModelFactory):
```
Comment (vtemian) [inline silver/tests/factories.py]: `TransactionProformaFactory`

Comment (bogdanpetrea) [inline silver/tests/factories.py]: Maybe use `TransactionFactory` like this: `TransactionFactory.create(invoice=None)`. Might need to make it work first.

**Pull request merged**